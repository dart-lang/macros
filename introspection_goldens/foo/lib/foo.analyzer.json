{
  "uris": {
    "package:foo/foo.dart": {
      "scopes": {
        "Foo": {
          "members": {
            "bar": {
              "properties": {
                "isAbstract": false,
                "isGetter": false,
                "isField": true,
                "isMethod": false,
                "isStatic": false
              }
            }
          }
        },
        "Bar": {
          "members": {
            "bar": {
              "properties": {
                "isAbstract": false,
                "isGetter": false,
                "isField": true,
                "isMethod": false,
                "isStatic": false
              }
            }
          }
        }
      }
    }
  },
  "types": {
    "named": {
      "package:foo/foo.dart#Foo": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "package:foo/foo.dart",
            "name": "Foo"
          },
          "instantiation": []
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      },
      "dart:core#Object": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "dart:core",
            "name": "Object"
          },
          "instantiation": []
        },
        "supertypes": []
      },
      "dart:core#Null": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "dart:core",
            "name": "Null"
          },
          "instantiation": []
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      },
      "dart:async#Future": {
        "typeParameters": [
          {
            "identifier": 0
          }
        ],
        "self": {
          "name": {
            "uri": "dart:async",
            "name": "Future"
          },
          "instantiation": [
            {
              "type": "TypeParameterTypeDesc",
              "value": {
                "parameterId": 0
              }
            }
          ]
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      },
      "package:foo/foo.dart#Bar": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "package:foo/foo.dart",
            "name": "Bar"
          },
          "instantiation": []
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      }
    }
  }
}