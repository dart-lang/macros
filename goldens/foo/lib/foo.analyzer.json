{
  "uris": {
    "package:foo/foo.dart": {
      "scopes": {
        "Foo": {
          "members": {
            "construct": {
              "properties": {
                "isAbstract": false,
                "isConstructor": true,
                "isGetter": false,
                "isField": false,
                "isMethod": false,
                "isStatic": false
              },
              "requiredPositionalParameters": [
                {
                  "type": "NamedTypeDesc",
                  "value": {
                    "name": {
                      "uri": "dart:core",
                      "name": "int"
                    },
                    "instantiation": []
                  }
                }
              ],
              "optionalPositionalParameters": [],
              "namedParameters": [
                {
                  "type": "NamedTypeDesc",
                  "value": {
                    "name": {
                      "uri": "dart:core",
                      "name": "int"
                    },
                    "instantiation": []
                  }
                }
              ]
            },
            "construct2": {
              "properties": {
                "isAbstract": false,
                "isConstructor": true,
                "isGetter": false,
                "isField": false,
                "isMethod": false,
                "isStatic": false
              },
              "requiredPositionalParameters": [
                {
                  "type": "NamedTypeDesc",
                  "value": {
                    "name": {
                      "uri": "dart:core",
                      "name": "int"
                    },
                    "instantiation": []
                  }
                }
              ],
              "optionalPositionalParameters": [
                {
                  "type": "NullableTypeDesc",
                  "value": {
                    "inner": {
                      "type": "NamedTypeDesc",
                      "value": {
                        "name": {
                          "uri": "dart:core",
                          "name": "String"
                        },
                        "instantiation": []
                      }
                    }
                  }
                }
              ],
              "namedParameters": []
            },
            "bar": {
              "properties": {
                "isAbstract": false,
                "isConstructor": false,
                "isGetter": false,
                "isField": true,
                "isMethod": false,
                "isStatic": false
              },
              "returnType": {
                "type": "NamedTypeDesc",
                "value": {
                  "name": {
                    "uri": "dart:core",
                    "name": "int"
                  },
                  "instantiation": []
                }
              }
            },
            "method": {
              "properties": {
                "isAbstract": false,
                "isConstructor": false,
                "isGetter": false,
                "isField": false,
                "isMethod": true,
                "isStatic": false
              },
              "returnType": {
                "type": "VoidTypeDesc"
              },
              "requiredPositionalParameters": [
                {
                  "type": "NamedTypeDesc",
                  "value": {
                    "name": {
                      "uri": "dart:core",
                      "name": "int"
                    },
                    "instantiation": []
                  }
                }
              ],
              "optionalPositionalParameters": [],
              "namedParameters": [
                {
                  "type": "NullableTypeDesc",
                  "value": {
                    "inner": {
                      "type": "NamedTypeDesc",
                      "value": {
                        "name": {
                          "uri": "dart:core",
                          "name": "int"
                        },
                        "instantiation": []
                      }
                    }
                  }
                }
              ]
            },
            "method2": {
              "properties": {
                "isAbstract": false,
                "isConstructor": false,
                "isGetter": false,
                "isField": false,
                "isMethod": true,
                "isStatic": false
              },
              "returnType": {
                "type": "NullableTypeDesc",
                "value": {
                  "inner": {
                    "type": "NamedTypeDesc",
                    "value": {
                      "name": {
                        "uri": "package:foo/foo.dart",
                        "name": "Bar"
                      },
                      "instantiation": []
                    }
                  }
                }
              },
              "requiredPositionalParameters": [
                {
                  "type": "NamedTypeDesc",
                  "value": {
                    "name": {
                      "uri": "dart:core",
                      "name": "int"
                    },
                    "instantiation": []
                  }
                }
              ],
              "optionalPositionalParameters": [
                {
                  "type": "NullableTypeDesc",
                  "value": {
                    "inner": {
                      "type": "NamedTypeDesc",
                      "value": {
                        "name": {
                          "uri": "dart:core",
                          "name": "String"
                        },
                        "instantiation": []
                      }
                    }
                  }
                }
              ],
              "namedParameters": []
            }
          },
          "properties": {
            "isClass": true
          }
        },
        "Bar": {
          "members": {
            "": {
              "properties": {
                "isAbstract": false,
                "isConstructor": true,
                "isGetter": false,
                "isField": false,
                "isMethod": false,
                "isStatic": false
              },
              "requiredPositionalParameters": [],
              "optionalPositionalParameters": [],
              "namedParameters": []
            },
            "bar": {
              "properties": {
                "isAbstract": false,
                "isConstructor": false,
                "isGetter": false,
                "isField": true,
                "isMethod": false,
                "isStatic": false
              },
              "returnType": {
                "type": "NamedTypeDesc",
                "value": {
                  "name": {
                    "uri": "dart:core",
                    "name": "int"
                  },
                  "instantiation": []
                }
              }
            }
          },
          "properties": {
            "isClass": true
          }
        }
      }
    }
  },
  "types": {
    "named": {
      "dart:core#Object": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "dart:core",
            "name": "Object"
          },
          "instantiation": []
        },
        "supertypes": []
      },
      "dart:core#Null": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "dart:core",
            "name": "Null"
          },
          "instantiation": []
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      },
      "dart:async#Future": {
        "typeParameters": [
          {
            "identifier": 0
          }
        ],
        "self": {
          "name": {
            "uri": "dart:async",
            "name": "Future"
          },
          "instantiation": [
            {
              "type": "TypeParameterTypeDesc",
              "value": {
                "parameterId": 0
              }
            }
          ]
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      },
      "package:foo/foo.dart#Foo": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "package:foo/foo.dart",
            "name": "Foo"
          },
          "instantiation": []
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      },
      "package:foo/foo.dart#Bar": {
        "typeParameters": [],
        "self": {
          "name": {
            "uri": "package:foo/foo.dart",
            "name": "Bar"
          },
          "instantiation": []
        },
        "supertypes": [
          {
            "name": {
              "uri": "dart:core",
              "name": "Object"
            },
            "instantiation": []
          }
        ]
      }
    }
  }
}